kong/init.lua

```diff
@@ -352,21 +390,42 @@ function Kong.init()
     certificate.init()
   end
 
-  mesh.init()
+  if kong.configuration.database ~= "off" then
+    mesh.init()
+  end
 
   -- Load plugins as late as possible so that everything is set up
   loaded_plugins = assert(db.plugins:load_plugin_schemas(config.loaded_plugins))
   sort_plugins_for_execution(config, db, loaded_plugins)
 
-  local err
-  plugins_map_semaphore, err = semaphore.new(1) -- 1 = treat this as a mutex
-  if not plugins_map_semaphore then
-    error("failed to create plugins map semaphore: " .. err)
+  do
+    local timeout = 60
+    if kong.configuration.database == "cassandra" then
+      -- cassandra_timeout is defined in ms
+      timeout = kong.configuration.cassandra_timeout / 1000
+
+    elseif kong.configuration.database == "postgres" then
+      -- pg_timeout is defined in ms
+      timeout = kong.configuration.pg_timeout / 1000
+    end
+    PLUGINS_MAP_MUTEX_OPTS = {
+      name = "plugins_map",
+      timeout = timeout,
+    }
   end
 
-  local _, err = build_plugins_map(db, "init")
-  if err then
-    error("error building initial plugins map: ", err)
+  if kong.configuration.database == "off" then
+    local err
+    declarative_entities, err = parse_declarative_config(config)
```

Comment:
Nitpick that will help future selves: please us `kong.configuration` here. Typically we use `kong_config` when we refer to the `kong.conf` parsed table, for clarity. For the sake of consistency (and in the same spirit of explicitness), we should keep avoiding using `config` nowdays.